src/figuresClasses.ts

```diff
@@ -1,11 +1,99 @@
-export interface Figure {}
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape = 'triangle';
+
+  color: string;
+
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(color: string, a: number, b: number, c: number) {
+    this.color = color;
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Side lengths must be more that zero');
+    }
+
+    const sortedSides = [a, b, c].sort((x, y) => x - y);
+
+    if (sortedSides[2] >= sortedSides[0] + sortedSides[1]) {
+      throw new Error(`Sides ${a}, ${b}, and ${c} can't form a triangle`);
+    }
+
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
```

Comment:
[CODE KNOWLEDGE] - You can simplify your constructor by removing the explicit assignments to class properties `this.a`, `this.b`, and `this.c` since TypeScript can automatically create class properties from constructor parameters with access modifiers.